src/configuration/app/source-operations.md

```diff
@@ -0,0 +1,74 @@
+---
+search:
+    keywords: ['.platform.app.yaml', 'automated', 'code', 'source', 'update', 'operation', 'dependencies', 'auto']
+---
+
+# Source operations
+
+> **note**
+>
+> Source Operations are currently in beta.  While the syntax is not expected to change some behavior might in the future.
+
+An application can define a number of operations that apply to its source code and that can be automated.
+
+A basic, common source operation could be to automatically update Composer dependencies like this:
+
+```yaml
+source:
+    operations:
+        update:
+            command: |
+                set -e
+                composer update
+                git add composer.lock
+                git commit -m "Update Composer dependencies."
+```
+
+The `update` key is the name of the operation. It is arbitrary, and multiple source operations can be defined.
+
+(You may with to include more robust error handling than this example.)
+
+The environment resource gets a new `source-operation` action which can be triggered by the CLI:
+
+```bash
+platform source-operation:run update
+```
+
+The `source-operation:run` command takes the command name to run plus, optionally, additional variables to inject in the environment of the source operation.  They will be interpreted the same way as any other [variable](/development/variables.md) set through the UI or CLI, which means you need an `env:` prefix to expose them as a Unix environment variable.  They can then be referenced by the source operation like any other variable.
+
+```bash
+platform source-operation:run update --variable env:FOO=bar --variable env:BAZ=beep
+```
+
+When this operation is triggered:
+
+* A clean Git checkout of the current environment HEAD commit is created; this checkout doesn't have any remotes, has all the tags defined in the project, but only has the current environment branch.
+* Sequentially, for each application that has defined this operation, the operation command is launched in the container image of the application.  The environment will have all of the variables available during the build phase, plus `PLATFORM_BRANCH` and `PLATFORM_ENVIRONMENT`, optionally overridden by the variables specified in the operation payload.
+* At the end of the process, if any commits were created, the new commits are pushed to the repository and the normal build process of the environment is triggered.
+
+Note that these operations run in an isolated container: it is not part of the runtime cluster of the environment, and doesn't require the environment to be running.  Also be aware that if multiple applications in a single project both result in a new commit, that will appear as two distinct commits in the Git history but only a single new build/deploy cycle will occur.
+
+## Automated Source Operations using cron
+
+You can use cron to automatically run your source operations.
```

Comment:
Damien thought that this should not be documented.

Reply:
He said that doing that was a bad idea, and as a result, Augustin took that part out of the How-to guide.

Reply:
But double check; it might be that the objection was solely based on having the CLI in the environment as a privileged user, and not because of the pattern of running source operations from cron. In that case, as long as we remind the user to have a service account for installing the CLI, then we're OK.

Reply:
Ugh.  I thought it was included here because it was safe.  We already tell people outright to add the CLI to all projects for production, so not doing it here seems dumb.

Reply:
It is safe to use cron to trigger source operations. What's not is to use your main API token to install the CLI on your app container.